extensions/azurecore/src/utils.ts

```diff
@@ -161,6 +166,118 @@ export async function updateTenantIgnoreList(tenantIgnoreList: string[]): Promis
 	await configuration.update(constants.Filter, tenantIgnoreList, vscode.ConfigurationTarget.Global);
 }
 
+export function updateProviderSettings(defaultSettings: ProviderSettings[]): ProviderSettings[] {
+	let providerSettingsJson: ProviderSettingsJson[] | undefined = vscode.workspace.getConfiguration(constants.AzureSection).get(constants.ProviderSettingsJson) as ProviderSettingsJson[];
+	vscode.workspace.onDidChangeConfiguration(async (changeEvent) => {
```

Comment:
How does this behavior if the user has auto-save on? Will it keep popping up the prompt every time they type a letter?

Reply:
it will pop up every time the file is saved yes, but the auto save setting itself will not save after every letter.  The options for autosave are onFocusChange, onWindowChange, or afterDelay (which can be set by the user).

Reply:
Ok, I would definitely suggest keeping an eye on this and having extra people test it out themselves to see if the experience is nice to use. It might be worth adding our own debounce delay to this with a longer timeout (especially if these don't take effect until after restart).

Telemetry could help here too if you wanted to add some events like how often the toast is shown...